# Самостоятельное задание #3


### Задание 1

Допишите функцию, которая принимает массив из 10 чисел и возвращает их сумму.

In [14]:
def SumArray(arr):
	sum = 0
	for i in arr:
		sum += i
	return sum

arr = [0, 1, 2, 3, 0, 4, 0]
print(SumArray(arr))

10



### Задание 2

Допишите функцию, которая принимает массив чисел и возвращает количество чисел равных нулю.

In [12]:
def CountZero(arr):
	count = 0
	for i in arr:
		if i == 0:
			count += 1
	return count

zero = [0, 1, 2, 3, 0, 4, 0]
print(CountZero(zero))

3



### Задание 3

Напишите программу, в которой задается  натуральное число n и выводится лестница из n ступенек, i-я ступенька должна состоять из чисел от 1 до i без пробелов.

In [15]:
x = 1
i = 4
while (x != i+1):
	for j in range(1, x+1):
		print(j, end="")
	x += 1
	print(" ")



1 
12 
123 
1234 


1
12
123
1234
12345


### Задание 4

Напишите программу, в которой задается натуральное число n и выводится пирамида из n ступенек, i-я ступень должна состоять из чисел от 1 до i и обратно без пробелов.

In [18]:
rows = 4
for i in range(1, rows + 1):
    for j in range(1, i - 1):
        print(j, end=" ")
    for j in range(i - 1, 0, -1):
        print(j, end=" ")
    print()


1 
1 2 1 
1 2 3 2 1 



### Задание 5

Напишите программу, в которой задается натуральное число n и выводится ромб из `n*2-1` ступенек, i-я ступень должна состоять из чисел от 1 до i и обратно без пробелов.

In [19]:
rows = 4
for i in range(1, rows + 1):
    for j in range(1, i - 1):
        print(j, end=" ")
    for j in range(i - 1, 0, -1):
        print(j, end=" ")
    print()
for i in range(rows, 0, -1):
    for j in range(1, i - 1):
        print(j, end=" ")
    for j in range(i - 1, 0, -1):
        print(j, end=" ")
    print()


1 
1 2 1 
1 2 3 2 1 
1 2 3 2 1 
1 2 1 
1 

